src/libdredd/src/mutate_visitor.cc

```diff
@@ -118,7 +118,7 @@ bool MutateVisitor::TraverseDecl(clang::Decl* decl) {
     return true;
   }
   if (const auto* var_decl = llvm::dyn_cast<clang::VarDecl>(decl)) {
-    if (var_decl->isConstexpr()) {
+    if (var_decl->isConstexpr() || var_decl->hasConstantInitialization()) {
       // Because Dredd's mutations occur dynamically, they cannot be applied to
       // C++ constexprs, which require compile-time evaluation.
```

Comment:
Can you please update the comment to reflect the additional check? I hadn't heard of this, but from [here](https://clang.llvm.org/docs/AttributeReference.html#require-constant-initialization-constinit-c-20
) it looks like `require_constant_initialization` attributes are supported by GCC and Clang, and that C++20 adds a `constinit` keyword.

Could you also add some tests that use the `constinit` keyword instead of the attribute? (Hopefully we can support C++20 tests in Dredd's single file tests.)

Reply:
Unfortunately because Dredd uses an earlier version of clang and the C++ standard, it throws the following error when trying to mutate a variable declaration using `constinit`:
```
/home/jlj/dev/dredd/temp/const_init_keyword.cc:2:3: error: unknown type name 'constinit'
  constinit static int x = 0;

```
This definitely feels like an issue to keep for future when we switch to using C++ 20.